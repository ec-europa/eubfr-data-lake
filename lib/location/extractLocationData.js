/**
 * Extracts the following properties for `project_locations:
 * - `country_code`
 * - `region`
 * - `town`
 *
 * @param {String} input
 * @returns {Object} results
 */
const extractLocationData = input => {
  if (!input) return {};

  const separator = new RegExp(/[,/]/i); // separate either by comma or slash.

  const results = {
    countryCodes: [],
    regions: [],
    towns: [],
    raw: input,
  };

  const locations = input
    .split(';')
    .map(p => p.trim())
    .filter(l => l);

  locations.forEach(location => {
    const infoArray = location
      .split(separator)
      // Improve the separate pieces of information for each location.
      .map(i => i.trim())
      .filter(i => i);

    // Country code is always the last item.
    const code = infoArray.pop();
    results.countryCodes.push(code);

    if (infoArray.length) {
      const [town, region] = infoArray;

      if (town) results.towns.push(town);
      if (region) results.regions.push(region);
    }
  });

  return results;
};

module.exports = extractLocationData;
