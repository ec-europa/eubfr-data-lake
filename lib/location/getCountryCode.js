// Helper to translate few specific NUTS0 codes to ISO 3166-1 Alpha-2 country codes

const getCountryCode = countryCode => {
  // Do nothing if falsy
  if (!countryCode) return countryCode;

  switch (countryCode.trim().toUpperCase()) {
    case 'UK':
      return 'GB';

    // Greece
    case 'EL':
      return 'GR';

    default:
      return countryCode;
  }
};

module.exports = getCountryCode;
