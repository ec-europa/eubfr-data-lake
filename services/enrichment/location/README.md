# Enrichment location

Enrich `project_locations`.
