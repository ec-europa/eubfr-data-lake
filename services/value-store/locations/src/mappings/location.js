module.exports = {
  location: {
    properties: {
      geolocation: {
        type: 'geo_shape',
      },
    },
  },
};
