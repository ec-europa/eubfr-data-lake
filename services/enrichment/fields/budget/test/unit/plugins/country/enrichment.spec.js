/**
 * @jest-environment node
 */

import plugin from '../../../../src/plugins/country';
import location from './location.json';

describe('Country enrichment plugin: depending on the following input', () => {
  test('Empty object: returns same object, untouched, no errors', async () => {
    const loc = {};
    const result = await plugin(loc);
    expect(result).toMatchSnapshot();
  });

  test('Object with a country_code property: returns a new object centroid data', async () => {
    const loc = JSON.parse(JSON.stringify(location));
    const result = await plugin(loc);
    expect(result).toMatchSnapshot();
  });
});
