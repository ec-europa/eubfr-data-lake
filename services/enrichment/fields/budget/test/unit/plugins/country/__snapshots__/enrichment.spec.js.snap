// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country enrichment plugin: depending on the following input Empty object: returns same object, untouched, no errors 1`] = `Object {}`;

exports[`Country enrichment plugin: depending on the following input Object with a country_code property: returns a new object centroid data 1`] = `
Object {
  "address": "",
  "centroid": Object {
    "lat": 50.6407351,
    "lon": 4.66696,
  },
  "country_code": "BE",
  "enriched": true,
  "location": Object {
    "coordinates": Array [
      4.66696,
      50.6407351,
    ],
    "type": "Point",
  },
  "nuts": Array [],
  "postal_code": "",
  "region": "",
  "town": "",
}
`;
